src/main/java/me/coley/analysis/util/TypeUtil.java

```diff
@@ -158,11 +158,14 @@ public static boolean isSubTypeOf(TypeChecker typeChecker, Type child, Type pare
 		if (bothArrays) {
 			// TODO: With usage cases of "isSubTypeOf(...)" should we just check the element types are equals?
 			//  - Or should sub-typing with array element types be used like it currently is?
-			if (child.getDimensions() != parent.getDimensions()) {
-				int minDims = Math.min(child.getDimensions(), parent.getDimensions());
-				// shitty hack
-				child = Type.getType(child.getDescriptor().substring(minDims));
-				parent = Type.getType(parent.getDescriptor().substring(minDims));
+			boolean dimsNe = child.getDimensions() != parent.getDimensions();
+
+			int minDims = Math.min(child.getDimensions(), parent.getDimensions());
+			// shitty hack
+			child = Type.getType(child.getDescriptor().substring(minDims));
+			parent = Type.getType(parent.getDescriptor().substring(minDims));
+
+			if (dimsNe) {
 				// dimensions dont match and parent sort isn't Object, this won't work
 				if (!parent.equals(TypeUtil.OBJECT_TYPE)) return false;
 			}
```

Comment:
```suggestion
  		// dimensions dont match and parent sort isn't Object, this won't work
			if (dimsNe && !parent.equals(TypeUtil.OBJECT_TYPE)) return false;
```
Don't you know about `&&`?

Reply:
Like I said on discord, I don't really care that much about the current state of the project as I don't use it. Sure it could be cleaned up. But if I needed to have an enhanced analyzer for ASM again, I'd just start over.